xbmc/utils/URIUtils.cpp

```diff
@@ -195,9 +195,12 @@ std::string URIUtils::GetFileName(const std::string& strFileNameAndPath)
     return GetFileName(url.GetFileName());
   }
 
-  /* find the last slash */
-  const size_t slash = strFileNameAndPath.find_last_of("/\\");
-  return strFileNameAndPath.substr(slash+1);
+  size_t lastPos = strFileNameAndPath.find_last_of(":/\\");
+  // If the colon is found at position 1, it is part of a Windows drive letter (e.g., "C:").
+  // Otherwise, it is treated as part of a URL or other path format.
+  if (lastPos != std::string::npos && strFileNameAndPath[lastPos] == ':' && lastPos != 1)
+    lastPos = std::string::npos;
```

Comment:
The handling of the `':'` isn't right here. If you encounter a `':'` and it's not part of the drive letter you have to continue to search for a separator. Additional unit test for this:
```cpp
    EXPECT_EQ("a:b", URIUtils::GetFileName("/hello/there/a:b"));
    EXPECT_EQ("a:b", CURL_FileName_URIUtils_Split("/hello/there/a:b"));
    EXPECT_EQ("a:b", URIUtils_Split("/hello/there/a:b"));
```